File: Modul-6/Soal-4/PRAK602-2410817110016-MuhammadAzriRahman.c
#include <stdio.h>
#include <string.h>

void main() {
    char kode[100];
    char pesan[100];
    char pesanEnkripsi[100];
    int bintang = 0;
    int pagar = 0;

    printf("Input\n");
    scanf("%[^\n] ", kode);
    scanf("%[^\n]", pesan);

    if (strlen(kode) != strlen(pesan)) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
    } else {
        for (int i = 0; i<strlen(kode); i++) {
            if (kode[i] == ' ' && pesan[i] == ' ') {
                pesanEnkripsi[i] = ' ';
            }
            else if (kode[i] == pesan[i]) {
                pesanEnkripsi[i] = '*';
                bintang++;
            } else {
                pesanEnkripsi[i] = '#';
                pagar++;
        } }
        pesanEnkripsi[strlen(pesan)] = '\0';

        printf("Output\n");
        printf("%s\n", pesanEnkripsi);
        printf("* = %d\n", bintang);
        printf("# = %d\n", pagar);
        
        if (bintang >= pagar) {
            printf("Pesan Asli\n");
        } else {
            printf("Pesan Palsu\n");
        }
    }
}

File: Modul-6/Soal-4/PRAK602-2410817110016-MuhammadAzriRahman.py
print("Input")
kode = input("")
pesan = input("")

pesanEnkripsi = []

bintang = 0
pagar = 0

if len(kode) != len(pesan):
    print('Panjang kalimat berbeda, pesan palsu')
else:
    for i in range(len(pesan)):
        if pesan[i] == ' ' and kode[i] == ' ':
            pesanEnkripsi.append(' ')
        elif pesan[i] == kode[i]:
            pesanEnkripsi.append('*')
            bintang += 1
        else:
            pesanEnkripsi.append('#')
            pagar += 1
    
    print("Output")
    print(f"{''.join(pesanEnkripsi)}")
    print(f"* = {bintang}")
    print(f"# = {pagar}")
    print("Pesan Asli") if bintang >= pagar else print("Pesan Palsu")